seonshine_frontend/src/components/organims/restaurantProfileModal/index.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import EditIcon from '@mui/icons-material/Edit';
+import EditOutlinedIcon from '@mui/icons-material/EditOutlined';
+import { Avatar, Box, Button, IconButton, Modal, Skeleton } from '@mui/material';
+
+import DatePicker from '@/components/molecules/datePicker/DatePicker';
+
+import { RestaurantDetailType } from '@/types/user';
+import { isValidImageFile } from '@/utils/file';
+
+import { useGetRestaurantDetailApi, useGetUserDetailApi } from '@/apis/hooks/userApi.hook';
+
+import { RestaurantInfoSchema, restaurantInfoSchema } from './schema';
+
+interface UserProfileModalProps {
+  userId: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const RestaurantProfileModal: React.FC<UserProfileModalProps> = ({ userId, isOpen, onClose }) => {
+  const { data: restaurant, isLoading } = useGetRestaurantDetailApi({ restaurant_id: userId });
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [uploadError, setUploadError] = useState<string | null>(null);
+  const [previewUrl, setPreviewUrl] = useState<string>(restaurant?.profile_picture_url || '');
+
+  const { control, handleSubmit, reset } = useForm({
+    defaultValues: {
+      username: restaurant?.username,
+      address: restaurant?.address,
+      phone_number: restaurant?.phone_number,
+    },
+    resolver: zodResolver(restaurantInfoSchema),
+  });
+
+  const handleEditToggle = () => setIsEditing(!isEditing);
+
+  const handleSave = (data: any) => {
+    // onSave({ ...user, ...data, profilePicture: previewUrl });
+    setIsEditing(false);
+  };
+
+  const handleCancel = () => {
+    setIsEditing(false);
+    setPreviewUrl(restaurant?.profile_picture_url || '');
+    reset({
+      username: restaurant?.username,
+      address: restaurant?.address,
+      phone_number: restaurant?.phone_number,
+    });
+  };
+
+  const handleClose = () => {
+    onClose();
```

Comment:
when closing and init component again. The state always init, we don't need to reset state